platform/os_windows.go

```diff
@@ -302,32 +303,101 @@ func restartHNS(ctx context.Context) error {
 	}
 	defer service.Close()
```

Comment:
Mostly I just want commetns about how

1) HNS should hot reload this. Restart is not an api

2) working on getting this into agentbaker

3) there  is a race if we set the reg ke then crash before restart.